runtime/lua/vim/fs.lua

```diff
@@ -609,4 +611,55 @@ function M.normalize(path, opts)
   return path
 end
 
+--- @param path string Path to remove
+--- @param ty string type of path
+--- @param recursive? boolean
+--- @param force? boolean
+local function rm(path, ty, recursive, force)
+  --- @diagnostic disable-next-line:no-unknown
+  local rm_fn
+
+  if ty == 'directory' then
+    if recursive then
+      for file, fty in vim.fs.dir(path) do
+        rm(M.joinpath(path, file), fty, true, force)
+      end
+    elseif not force then
+      error(string.format('%s is a directory', path))
+    end
+
+    rm_fn = uv.fs_rmdir
+  else
+    rm_fn = uv.fs_unlink
+  end
+
+  local ret, err, errnm = rm_fn(path)
+  if ret == nil and (not force or errnm ~= 'ENOENT') then
+    error(err)
+  end
+end
+
+--- @class vim.fs.rm.Opts
+--- @inlinedoc
+---
+--- Remove directories and their contents recursively
+--- @field recursive? boolean
+---
+--- Ignore nonexistent files and arguments
+--- @field force? boolean
```

Comment:
what's meant by the "and arguments" part? 

should we make `force` behavior the default? it's probably the common intention, so would save some toil. it is the default in https://github.com/jprichardson/node-fs-extra/blob/master/docs/remove.md for example.

trying to delete a non-existent file doesn't seem like an error-worthy situation by default.

Reply:
I've tried to match the semantics of the shell `rm` command as much as possible in order to not create any new patterns. Anyone familiar with `rm` will be familiar with `vim.fs.rm()`. These field descriptions are copied from the [man page](https://man7.org/linux/man-pages/man1/rm.1.html) of coreutils `rm`.

Reply:
It's good to use familiar tools as a baseline, but `vim.fs` is also positioned as a high-level, ergonomic interface. We don't have a requirement to match shell tools.

Reply:
`rm` is a high level interface as it gets; something a user literally has to type and execute by hand, and does regularly.

We don't have a _requirement_ to match shell tools, I just think it's a good idea, especially with one that's as simple and as prolific as `rm` and deviating from that in any way has a mental cost.

Reply:
> as prolific as `rm` and deviating from that in any way has a mental cost.

fs-extra is downloaded 93 million times per week: https://www.npmjs.com/package/fs-extra

people prefer the "don't bug me" DWIM behavior. and because Nvim is an editor, DWIM is a central tenet.

Reply:
Nothing indicates the popularity of that library because of the `remove()` function. It's probably a dependency in a few js frameworks that probably make up a good portion of that number.

I could similarly correlate how much people like the `rm` command to how much Linux is installed, which would be much greater than 93 million.